vrt/ci.js

```diff
@@ -0,0 +1,351 @@
+      head: SNAPSHOT_BRANCH,
+      base: BUILDKITE_BRANCH,
+      body:
+        `This PR was generated based on visual changes detected in #${BUILDKITE_PULL_REQUEST}. ` +
+        `Please verify that the updated snapshots look correct before merging this PR into \`${BUILDKITE_BRANCH}\`.`,
+    });
+    _echo(`Created a new snapshot PR: ${pullRequest.data.html_url}`);
+    return pullRequest.data;
+  } catch (er) {
+    _echo(`There was an error creating a new snapshot PR.`);
+    _echo(er);
+    _exit(1);
+  }
+}
+
+async function getSnapshotPullRequest() {
+  try {
+    const pullRequests = await octokit.pulls.list({
+      owner: `uber`,
+      repo: `baseweb`,
+      head: `uber:${SNAPSHOT_BRANCH}`,
+    });
+    const pullRequest = pullRequests.data[0]; // should only ever be one PR
+    if (pullRequest) {
+      _echo(`An exisitng snapshot PR was found.`);
+      return pullRequest;
+    } else {
+      _echo(`No exisitng snapshot PR was found.`);
+      return null;
+    }
+  } catch (er) {
+    _echo(`There was an error fetching existing PRs.`);
+    _echo(`⮑ Could not find an existing snapshot PR.`);
+    _echo(er);
+    _exit(1);
+  }
+}
+
+function pushChangesToGitHub() {
+  _echo(
+    `Creating a new snapshot branch: ${SNAPSHOT_BRANCH}. ` +
+      `This will overwrite any existing snapshot branch.`,
+  );
+  _exec(`git checkout -b ${SNAPSHOT_BRANCH}`);
+  _exec(`git add vrt/__image_snapshots__/`);
+  _echo(`Commiting updated snapshots to ${SNAPSHOT_BRANCH}.`);
+  _exec(
+    `git commit -m "test(vrt): update visual snapshots for ${SHORT_BASE_COMMIT_HASH} [ci skip]"`,
+  );
+  _echo(`Force pushing updated snapshot branch to GitHub.`);
+  _exec(`git push --force origin ${SNAPSHOT_BRANCH}`);
+}
+
+function configureGit() {
+  _echo(`Configuring git to allow for pushing new commits & branches.`);
+  _exec(
+    `git config --global url."https://${GITHUB_BOT_AUTH_TOKEN}:@github.com/".insteadOf "https://github.com/"`,
+  );
+  _exec(`git config --global user.email ${GITHUB_BOT_EMAIL}`);
+  _exec(`git config --global user.name ${GITHUB_BOT_NAME}`);
+}
+
+function runTestsWithUpdates() {
+  _echo(`Running visual snapshot tests with updates.`);
+  _exec(`yarn vrt -u`);
+}
+
+function someSnapshotsWereUpdated() {
+  return new Promise(resolve => {
+    _exec(`git status --porcelain`, (code, stdout, stderr) => {
+      const changedFiles = stdout.split(`\n`);
+      const updatedSnapshots = changedFiles.filter(s =>
+        s.match(/vrt\/__image_snapshots__\//),
+      );
+      const result = updatedSnapshots.length > 0;
+      if (result) {
+        _echo(`Some snapshots were updated.`);
+      } else {
+        _echo(`No snapshots were updated.`);
+      }
+      resolve(result);
+    });
+  });
+}
+
+// A few wrappers around shell commands
+function _echo(message) {
+  return shell.echo(`❖ VRT: ${message}`);
+}
+
+function _exec(command, callback) {
+  return shell.exec(command, {silent: true}, callback);
```

Comment:
instead of shelljs, can we use https://nodejs.org/dist/latest-v10.x/docs/api/child_process.html#child_process_child_process_execsync_command_options?

that way, we wouldn't need the extra dependency

Reply:
oh i didn't know about this - yeah i think it can all be replaced then.